### ISSUE TITLE
Geopandas integration

### ISSUE THREAD

**Issue by Casyfill** [mapbox/mapboxgl-jupyter#66]
it would be great to have a geopandas integration.
I can envision it working like this (tqdm-pandas style):

```
import mapboxgl as mbx
import geopandas as gp
mbx.mapboxgl()  # initialisation - checking serial number and adding function to geodataFrame

gdf = gp.read_file('points.geojson')
gdf.mbx_plot(figsize=(18,10), column='Avg Medicare Payments', cmap='...')
```

it  would be awesome ( in my opinion ) to mimic matplotlib-esque interface in this case as much as it is possible - all the way to 
```
ax = gdf.mbx_plot(figsize=(18,10)
gdf2.mbx_plot(ax=ax)  # second layer
```

**Comment by Casyfill** [comment]
what do you think? I might try to make a draft PR

**Comment by ryanbaumann** [comment]
@caseyfill absolutely!  Currently the MapboxViz class takes `data` passed to the class initializer as a geojson format string, or URL to a geojson format object.  

In that light, I would consider just making a geopandas example workbook that directly passes a geodataframe to a viz and converts to geojson on the fly.  Would that make sense based on your experience, or should we consider directly accepting `data` as a geopandas dataframe?

**Comment by ryanbaumann** [comment]
@Casyfill is this something you're planning to PR?  Putting this on the target for our end of April release, 0.7.0

**Comment by anmolgarg** [comment]
This works well for me.
```
import json
import geopandas

if isinstance(data, geopandas.geodataframe.GeoDataFrame):
    data = json.loads(data.to_json())
```

**Comment by Casyfill** [comment]
sure, it can work, just looks ugly :-)



<https://mailtrack.io/> Sent with Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality&>
